Source/x11/XGServerEvent.m

```diff
@@ -1243,18 +1244,24 @@ - (void) processEvent: (XEvent *) event
            * it's not one of ours.
            * If it has gone to our root window - use the icon window.
            * If it has gone to a window - we see if it is one of ours.
+           * If it has gone to our unmapped window - deactivate.
            */
           XGetInputFocus(xEvent.xfocus.display, &fw, &rev);
           NSDebugLLog(@"NSEvent", @"%lu FocusOut\n",
                       xEvent.xfocus.window);
-          if (fw != None && fw != PointerRoot)
+          if (cWin && (fw == cWin->parent) && (cWin->map_state != IsViewable))
```

Comment:
Using cwin here is most likely wrong. cwin is a #define on generic.cachedWindow. Which just means the last window that was used for this XGServer. It is a pure coincidence when the focus goes to the parent of the last window for which we did process an event. You seem to expect that the last event was a previous FocusIn event and that cwin is still pointing to that. But this may not be true.

Reply:
This check is for case when the window of application is closing. The next window that receives focus in WindowMaker is window decorations (titlebar, resizebar). So `cWin` at this point contains window with content but in unmapped state. I've added this check to prevent application deactivation and focus loss.
If this is not such situation - fine, `else if` code block executed (old behavior). As I said before I observed focus loss during workspace switch in WindowMaker. 

In any case this code should be harmless in most cases, except the described above.

Reply:
But why do you think that cWin is the unmapped content window? It might be in some cases but there is nothing in the code that ensures that.
We should not build our code on coincidences.

Reply:
Because these 2 events are going one after other. If it's not - this is not the case I've talked about.

Reply:
I need some testing to recall what this change is for.

Reply:
While testing I observe situations when this code block was ran. But I can't clearly describe it now. Removed that part of code for now. I'll back to it later maybe.